src/Microsoft.AspNetCore.All/build/netcoreapp2.1/Microsoft.AspNetCore.All.targets

```diff
@@ -2,6 +2,6 @@
   <Target Name="EnsureTFMCompatibility">
     <Error
       Text="This version of Microsoft.AspNetCore.All is only compatible with the netcoreapp2.1 target framework."
-      Condition="'$(TargetFramework)' != 'netcoreapp2.1'"/>
+      Condition="'$(TargetFramework)' != 'netcoreapp2.1' AND '$(IsCrossTargetingBuild)' != 'true'"/>
```

Comment:
This concerns me a little. It means that you can bypass this check by using <SharedFrameworks> instead of <SharedFramework>. We might need a different mechanism to key off of.

Reply:
What about `Condition=" '$(TargetFramework)' != 'netcoreapp2.1' AND '$(TargetFramework)' != '' "`?

Reply:
I'll give that a try.